flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/abilities/source/SourceAbilitySpec.java

```diff
@@ -56,4 +57,12 @@
      */
     @JsonIgnore
     Optional<RowType> getProducedType();
+
+    /**
+     * When Filter,Limit,Partition,etc. are pushed into {@link DynamicTableSource}, the source will
+     * add a digest to record this behavior.
+     *
+     * @param exprStrs The expressions need to be added into the extra digest.
+     */
+    String getExtraDigests(Optional<List<String>> exprStrs);
```

Comment:
Having to pass an argument seems like a violation of the intention of the change. If the spec is responsible for generating the digest, it shouldn't need additional information to do so.